### ISSUE TITLE
cannot set own full name or email address

### ISSUE THREAD

**Issue by individual-it** [owncloud/core#30915]
### Steps to reproduce
1. login as user
2. navigate to personal general settings page
3. change full name or email address

### Expected behaviour
change should work

### Actual behaviour
![image](https://user-images.githubusercontent.com/2425577/37904446-835f23aa-311b-11e8-9351-40cc7ba1a9e1.png)

### Server configuration
**Operating system**:
Ubuntu 17.10
**Web server:**
Apache2
**Database:**
Sqlite
**PHP version:**
7.1
**ownCloud version:** (see ownCloud admin page)
10.1 prealpha
**Updated from an older ownCloud or fresh install:**
fresh
**Where did you install ownCloud from:**
git

**The content of config/config.php:**

```
{
    "system": {
        "license-key": "***REMOVED SENSITIVE VALUE***",
        "instanceid": "oc35rjhx6uol",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost"
        ],
        "datadirectory": "\/home\/artur\/www\/owncloud-core\/data",
        "overwrite.cli.url": "http:\/\/localhost\/owncloud-core",
        "dbtype": "sqlite3",
        "version": "10.1.0.0",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "loglevel": 0,
        "maintenance": false,
        "default_language": "en",
        "skeletondirectory": "\/home\/artur\/www\/owncloud-core\/apps\/notifications\/tests\/acceptance\/..\/..\/..\/..\/tests\/acceptance\/skeleton",
        "csrf.disabled": true,
        "enable_previews": false,
        "theme": "",
        "files_external_allow_create_new_local": "true",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "1025",
        "singleuser": false
    }
}

```

**List of activated apps:**

```
Enabled:
  - comments: 0.3.0
  - dav: 0.3.2
  - diagnostics: 0.1.0
  - enterprise_key: 0.1.4
  - federatedfilesharing: 0.3.1
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.7.1
  - files_sharing: 0.10.1
  - files_texteditor: 2.2.1
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - firewall: 2.5.0
  - gallery: 16.0.2
  - market: 0.2.3
  - notifications: 0.3.3
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - updatenotification: 0.2.1
Disabled:
  - customgroups
  - files_antivirus
  - multidirtest
  - password_policy
  - security
  - testing
  - theme-example
  - user_ldap
  - user_management

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
local
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
FF 59
**Operating system:**
Ubuntu 17.10 
#### Browser log
setting full name:
![image](https://user-images.githubusercontent.com/2425577/37904568-f1773896-311b-11e8-9093-f1a654c973b5.png)

setting email address:
two requests are send, both with `mailAddress: xxxx@aaaa.de` as parameter
![image](https://user-images.githubusercontent.com/2425577/37904614-147ffc06-311c-11e8-84af-c4f0809d74d6.png)

![image](https://user-images.githubusercontent.com/2425577/37904647-29ae05aa-311c-11e8-8640-fe408d230121.png)

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks the contributor most likely able to help you is @phil-davis.

Possibly related issues are https://github.com/owncloud/core/issues/16057 (Not possible to remove email address from personal info), https://github.com/owncloud/core/issues/27626 (Can't set the same email to two users), https://github.com/owncloud/core/issues/27546 (Cannot set e-mail address), https://github.com/owncloud/core/issues/18335 (Email address can't be set or changed for external user ), and https://github.com/owncloud/core/issues/27916 (Can't test Email server settings, but can't set personal email address either).

**Comment by phil-davis** [comment]
Works OK in ``stable10``

Broken in ``master``

**Comment by individual-it** [comment]
@DeepDiver1975  looks like it was introduced #30673

**Comment by individual-it** [comment]
problem in https://github.com/owncloud/core/blob/master/settings/js/panels/profile.js#L37
it tries to use the old private settings API
need to use the provisioning API instead, so it also works when

**Comment by individual-it** [comment]
fix in #30920
